### PR TITLE
[ROCKETMQ-18]Clean code

### PR DESCRIPTION
**Pull request by dongeforever**
Clean code

@vongosling @zhouxinyu please take a time to review it.

### CODE REVIEW

**Comment by zhouxinyu** [comment]
It seems ok for me. @lollipopjin  please review this pr.

**Comment by lizhanhui** [comment]
@zhouxinyu This PR introduces grammar of Java 7, say diamond. Do you plan to upgrade the language compatible version?

**Comment by lollipopjin** [comment]
@lizhanhui  
`      
       

        <maven.compiler.source>1.7</maven.compiler.source>

        <maven.compiler.target>1.7</maven.compiler.target>
` Already

File: client/src/main/java/org/apache/rocketmq/client/impl/MQClientAPIImpl.java [hunk 1]
```diff
@@ -469,6 +467,7 @@ private void onExceptionImpl(final String brokerName, //
             try {
                 sendCallback.onException(e);
             } catch (Exception e2) {
+                log.error("", e2);
```
Comment (zhouxinyu) [inline client/src/main/java/org/apache/rocketmq/client/impl/MQClientAPIImpl.java]: // Ignore

File: client/src/main/java/org/apache/rocketmq/client/impl/factory/MQClientInstance.java [hunk 2]
```diff
@@ -1134,10 +1131,10 @@ public ConsumerRunningInfo consumerRunningInfo(final String consumerGroup) {
 
         List<String> nsList = this.mQClientAPIImpl.getRemotingClient().getNameServerAddressList();
 
-        StringBuffer strBuffer = new StringBuffer();
+        StringBuilder strBuffer = new StringBuilder();
```
Comment (zhouxinyu) [inline client/src/main/java/org/apache/rocketmq/client/impl/factory/MQClientInstance.java]: use strBuilder

File: store/src/main/java/org/apache/rocketmq/store/CommitLog.java [hunk 3]
```diff
@@ -1150,15 +1150,15 @@ public AppendMessageResult doAppend(final long fileFromOffset, final ByteBuffer
...
-            final short propertiesLength = propertiesData == null ? 0 : (short) propertiesData.length;
+            final int propertiesLength = propertiesData == null ? 0 : propertiesData.length;
 
             if (propertiesLength > Short.MAX_VALUE) {
                 log.warn("putMessage message properties length too long. length={}", propertiesData.length);
                 return new AppendMessageResult(AppendMessageStatus.PROPERTIES_SIZE_EXCEEDED);
             }
 
             final byte[] topicData = msgInner.getTopic().getBytes(MessageDecoder.CHARSET_UTF8);
-            final int topicLength = topicData == null ? 0 : topicData.length;
```
Comment (zhouxinyu) [inline store/src/main/java/org/apache/rocketmq/store/CommitLog.java]: topicData may be null.

File: broker/src/main/java/org/apache/rocketmq/broker/filtersrv/FilterServerUtil.java [hunk 4]
```diff
@@ -26,9 +26,9 @@ public static void callShell(final String shellString, final Logger log) {
             String[] cmdArray = splitShellString(shellString);
             process = Runtime.getRuntime().exec(cmdArray);
             process.waitFor();
-            log.info("callShell: <{}> OK", shellString);
+            log.info("CallShell: <{}> OK", shellString);
         } catch (Throwable e) {
-            log.error("callShell: readLine IOException, " + shellString, e);
+            log.error("CallShell: readLine IOException, {}", shellString, e);
```
Comment (vongosling) [inline broker/src/main/java/org/apache/rocketmq/broker/filtersrv/FilterServerUtil.java]: Wrong usage for log.error, please review it

File: broker/src/main/java/org/apache/rocketmq/broker/slave/SlaveSynchronize.java [hunk 5]
```diff
@@ -68,10 +68,10 @@ private void syncTopicConfig() {
                         .putAll(topicWrapper.getTopicConfigTable());
                     this.brokerController.getTopicConfigManager().persist();
 
-                    log.info("update slave topic config from master, {}", masterAddrBak);
+                    log.info("Update slave topic config from master, {}", masterAddrBak);
                 }
             } catch (Exception e) {
-                log.error("syncTopicConfig Exception, " + masterAddrBak, e);
+                log.error("SyncTopicConfig Exception, {}", masterAddrBak, e);
```
Comment (vongosling) [inline broker/src/main/java/org/apache/rocketmq/broker/slave/SlaveSynchronize.java]: Wrong usage

File: broker/src/main/java/org/apache/rocketmq/broker/slave/SlaveSynchronize.java [hunk 6]
```diff
@@ -85,9 +85,9 @@ private void syncConsumerOffset() {
                 this.brokerController.getConsumerOffsetManager().getOffsetTable()
                     .putAll(offsetWrapper.getOffsetTable());
                 this.brokerController.getConsumerOffsetManager().persist();
-                log.info("update slave consumer offset from master, {}", masterAddrBak);
+                log.info("Update slave consumer offset from master, {}", masterAddrBak);
             } catch (Exception e) {
-                log.error("syncConsumerOffset Exception, " + masterAddrBak, e);
+                log.error("SyncConsumerOffset Exception, {}", masterAddrBak, e);
```
Comment (vongosling) [inline broker/src/main/java/org/apache/rocketmq/broker/slave/SlaveSynchronize.java]: Wrong

File: broker/src/main/java/org/apache/rocketmq/broker/slave/SlaveSynchronize.java [hunk 7]
```diff
@@ -106,12 +106,12 @@ private void syncDelayOffset() {
                     try {
                         MixAll.string2File(delayOffset, fileName);
                     } catch (IOException e) {
-                        log.error("persist file Exception, " + fileName, e);
+                        log.error("Persist file Exception, {}", fileName, e);
```
Comment (vongosling) [inline broker/src/main/java/org/apache/rocketmq/broker/slave/SlaveSynchronize.java]: wrong

File: broker/src/main/java/org/apache/rocketmq/broker/slave/SlaveSynchronize.java [hunk 8]
```diff
@@ -106,12 +106,12 @@ private void syncDelayOffset() {
...
                     }
                 }
-                log.info("update slave delay offset from master, {}", masterAddrBak);
+                log.info("Update slave delay offset from master, {}", masterAddrBak);
             } catch (Exception e) {
-                log.error("syncDelayOffset Exception, " + masterAddrBak, e);
+                log.error("SyncDelayOffset Exception, {}", masterAddrBak, e);
```
Comment (vongosling) [inline broker/src/main/java/org/apache/rocketmq/broker/slave/SlaveSynchronize.java]: wrong

File: broker/src/main/java/org/apache/rocketmq/broker/slave/SlaveSynchronize.java [hunk 9]
```diff
@@ -134,10 +134,10 @@ private void syncSubscriptionGroupConfig() {
                     subscriptionGroupManager.getSubscriptionGroupTable().putAll(
                         subscriptionWrapper.getSubscriptionGroupTable());
                     subscriptionGroupManager.persist();
-                    log.info("update slave Subscription Group from master, {}", masterAddrBak);
+                    log.info("Update slave Subscription Group from master, {}", masterAddrBak);
                 }
             } catch (Exception e) {
-                log.error("syncSubscriptionGroup Exception, " + masterAddrBak, e);
+                log.error("SyncSubscriptionGroup Exception, {}", masterAddrBak, e);
```
Comment (vongosling) [inline broker/src/main/java/org/apache/rocketmq/broker/slave/SlaveSynchronize.java]: wrong

File: client/src/main/java/org/apache/rocketmq/client/impl/consumer/DefaultMQPullConsumerImpl.java [hunk 10]
```diff
@@ -228,6 +228,7 @@ public void subscriptionAutomatically(final String topic) {
                     topic, SubscriptionData.SUB_ALL);
                 this.rebalanceImpl.subscriptionInner.putIfAbsent(topic, subscriptionData);
             } catch (Exception e) {
+                log.error("", e);
```
Comment (vongosling) [inline client/src/main/java/org/apache/rocketmq/client/impl/consumer/DefaultMQPullConsumerImpl.java]: So what would you do if you had all the security in the world?

File: client/src/main/java/org/apache/rocketmq/client/impl/consumer/DefaultMQPullConsumerImpl.java [hunk 11]
```diff
@@ -247,6 +248,7 @@ public void executeHookBefore(final ConsumeMessageContext context) {
                 try {
                     hook.consumeMessageBefore(context);
                 } catch (Throwable e) {
+                    log.error("", e);
```
Comment (vongosling) [inline client/src/main/java/org/apache/rocketmq/client/impl/consumer/DefaultMQPullConsumerImpl.java]: So what would you do if you had all the security in the world?

File: client/src/main/java/org/apache/rocketmq/client/impl/consumer/DefaultMQPullConsumerImpl.java [hunk 12]
```diff
@@ -258,6 +260,7 @@ public void executeHookAfter(final ConsumeMessageContext context) {
                 try {
                     hook.consumeMessageAfter(context);
                 } catch (Throwable e) {
+                    log.error("", e);
```
Comment (vongosling) [inline client/src/main/java/org/apache/rocketmq/client/impl/consumer/DefaultMQPullConsumerImpl.java]: So what would you do if you had all the security in the world?

**Comment by vongosling** [comment]
That's ok

**Pull request closed without merging**